devicetypes/heltun/heltun-hls01-thermostat.src/heltun-hls01-thermostat.groovy

```diff
@@ -0,0 +1,437 @@
+			map.name = "voltage"
+			map.value = Math.round(cmd.scaledMeterValue)
+			map.unit = "V"
+			sendEvent(map)
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatoperatingstatev2.ThermostatOperatingStateReport cmd) {
+	def state = (cmd.operatingState == 1) ? "heating" : "idle" //DeviceScale
+	sendEvent(name: "thermostatOperatingState", value: state)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatsetpointv2.ThermostatSetpointReport cmd) {   
+	def locaScale = getTemperatureScale() //HubScale
+	def deviceScale = (cmd.scale == 1) ? "F" : "C" //DeviceScale
+	def deviceTemp = cmd.scaledValue
+	def setPoint = (deviceScale == locaScale) ? deviceTemp : (deviceScale == "F" ? roundC(fahrenheitToCelsius(deviceTemp)) : celsiusToFahrenheit(deviceTemp).toDouble().round(0).toInteger())
+	def mode = modeToNumMap[device.currentValue("thermostatMode")]
+	if (mode == 0) {setPoint = 0}    
+	sendEvent(name: "heatingSetpoint", value: setPoint, unit: locaScale)   
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatmodev2.ThermostatModeSupportedReport cmd) {
+	def tSupportedModes = []
+	if(cmd.heat) { tSupportedModes << "heat" }
+	if(cmd.autoChangeover) { tSupportedModes << "autochangeover" }
+	if(cmd.dryAir) { tSupportedModes << "dryair" }
+	if(cmd.energySaveHeat) { tSupportedModes << "energysaveheat" }
+	if(cmd.away) { tSupportedModes << "away" }
+	if(cmd.off) { tSupportedModes << "off" }    
+	state.supportedModes = tSupportedModes
+	sendEvent(name: "supportedThermostatModes", value: tSupportedModes, displayed: false)
+}
+
+def setHeatingSetpoint(tValue) {
+	def cmds = []
+	def mode = device.currentValue("thermostatMode")
+	def currentMode = modeToNumMap[mode]
+	def temp = state.heatingSetpoint = tValue.toDouble() //temp got fromm the app
+	def tempInC = (getTemperatureScale() == "F" ? roundC(fahrenheitToCelsius(temp)) : temp) //If not C, Convert to C     
+	cmds << zwave.thermostatSetpointV2.thermostatSetpointSet(setpointType: currentMode, scale: 0, precision: 1, scaledValue: tempInC).format()
+
+	// Sync temp, opState, setPoint
+	cmds << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType:1).format()
+	cmds << zwave.thermostatOperatingStateV2.thermostatOperatingStateGet().format()
+	cmds << zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: currentMode).format()
+	sendHubCommand(cmds)
+}
+
+def setThermostatMode(String value) {
+	def cmds = []
+	cmds << zwave.thermostatModeV2.thermostatModeSet(mode: modeToNumMap[value]).format()
+	cmds << zwave.thermostatModeV2.thermostatModeGet().format()
+	cmds << zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: modeToNumMap[value]).format()
+	sendHubCommand(cmds)
+}
+
+def getNumToModeMap() {
+[   
+	0 : "off",
+	1 : "heat",
+	10 : "autochangeover",
+	13 : "away",
+	11 : "energysaveheat",
+	8 : "dryair"   
+]}
+
+def getModeToNumMap() {
+[
+	"off": 0,
+	"heat": 1,
+	"autochangeover": 10,
+	"away": 13,
+	"energysaveheat": 11,
+	"dryair": 8 
+]}
```

Comment:
Please sort this map by value and fix indentations (+1 tab) and move `}` to the next line.